src/compiler/transformers/declarations/localInferenceResolver.ts

```diff
@@ -0,0 +1,879 @@
+import {
+    addRelatedInfo,
+    ArrayLiteralExpression,
+    ArrowFunction,
+    AsExpression,
+    BindingElement,
+    ComputedPropertyName,
+    createDiagnosticForNode,
+    createPropertyNameNodeForIdentifierOrLiteral,
+    Debug,
+    DiagnosticMessage,
+    Diagnostics,
+    DiagnosticWithLocation,
+    EntityNameOrEntityNameExpression,
+    ExportAssignment,
+    findAncestor,
+    FunctionDeclaration,
+    FunctionExpression,
+    GetAccessorDeclaration,
+    getCommentRange,
+    getEmitScriptTarget,
+    getMemberKeyFromElement,
+    getNameOfDeclaration,
+    getTextOfNode,
+    HasInferredType,
+    hasSyntacticModifier,
+    Identifier,
+    isAccessor,
+    isAsExpression,
+    isBinaryExpression,
+    isClassExpression,
+    isComputedPropertyName,
+    isConstTypeReference,
+    isEntityNameExpression,
+    isExpandoPropertyDeclaration,
+    isExportAssignment,
+    isGetAccessor,
+    isIdentifier,
+    isInterfaceDeclaration,
+    isLiteralTypeNode,
+    isMethodDeclaration,
+    isMethodOrAccessor,
+    isNoSubstitutionTemplateLiteral,
+    isNumericLiteral,
+    IsolatedTransformationContext,
+    isOmittedExpression,
+    isOptionalDeclaration,
+    isParameter,
+    isParenthesizedExpression,
+    isPrefixUnaryExpression,
+    isPrivateIdentifier,
+    isPropertyAssignment,
+    isPropertyDeclaration,
+    isPropertyName,
+    isPropertySignature,
+    isSetAccessor,
+    isShorthandPropertyAssignment,
+    isSpreadAssignment,
+    isSpreadElement,
+    isStatement,
+    isStringDoubleQuoted,
+    isStringLiteral,
+    isStringLiteralLike,
+    isThisIdentifier,
+    isTypeAssertionExpression,
+    isTypeLiteralNode,
+    isTypeNode,
+    isTypeParameterDeclaration,
+    isTypeReferenceNode,
+    isUnionTypeNode,
+    isVariableDeclaration,
+    KeywordTypeSyntaxKind,
+    length,
+    LiteralExpression,
+    MethodDeclaration,
+    ModifierFlags,
+    Node,
+    NodeArray,
+    NodeFlags,
+    nullTransformationContext,
+    ObjectLiteralExpression,
+    ParameterDeclaration,
+    ParenthesizedExpression,
+    PrefixUnaryExpression,
+    PropertyDeclaration,
+    PropertyName,
+    PropertySignature,
+    SetAccessorDeclaration,
+    setCommentRange,
+    setTextRange,
+    ShorthandPropertyAssignment,
+    SourceFile,
+    SpreadAssignment,
+    SpreadElement,
+    Symbol,
+    SyntaxKind,
+    TransformationContext,
+    TypeAssertion,
+    TypeElement,
+    TypeNode,
+    unescapeLeadingUnderscores,
+    VariableDeclaration,
+    visitEachChild,
+    visitNode,
+    visitNodes,
+    Visitor,
+    VisitResult,
+} from "../../_namespaces/ts";
+
+enum NarrowBehavior {

```

Comment:
Likely this should be declared as a `const enum`, though for the esbuild produced bundle it will still be fast.